src/Analyzer.JsonRuleEngine/JsonRuleEngine.cs

```diff
@@ -42,11 +43,30 @@ public JsonRuleEngine(BuildIJsonLineNumberResolver jsonLineNumberResolverBuilder
         /// <returns>The results of the rules against the template.</returns>
         public IEnumerable<IEvaluation> EvaluateRules(TemplateContext templateContext, string ruleDefinitions)
         {
-            List<RuleDefinition> rules = JsonConvert.DeserializeObject<List<RuleDefinition>>(ruleDefinitions);
+            List<RuleDefinition> rules;
 
-            foreach(RuleDefinition rule in rules)
+            try
             {
-                var ruleExpression = rule.ExpressionDefinition.ToExpression(BuildLineNumberResolver(templateContext));
+                rules = JsonConvert.DeserializeObject<List<RuleDefinition>>(ruleDefinitions);
+            }
+            catch (Exception e)
+            {
+                throw new JsonRuleEngineException($"Failed to parse rules {e.Message}");
```

Comment:
For these exceptions, instead of appending `e.Message`, add `e` as the `innerException`.